gobblin-utility/src/main/java/gobblin/util/RateControlledFileSystem.java

```diff
@@ -139,8 +141,24 @@ public boolean mkdirs(Path path, FsPermission permission) throws IOException {
 
   @Override
   public boolean rename(Path path0, Path path1) throws IOException {
+    boolean renamed;
+
     this.acquirePermit();
-    return this.fs.rename(path0, path1);
+
+    // Use Java rename when renaming a directory for LocalFileSystem.
```

Comment:
This code appears in two places. Please consider refactoring it into a function.